internal/job/restore/restore_test.go

```diff
@@ -833,9 +834,215 @@ func TestRestore_doRestoreRawImagePhase_skip(t *testing.T) {
+	metadata, err := job.GetBackupMetadata(cfg.finRepo)
+	require.NoError(t, err)
+	require.NoError(t, r.doRestoreDiffPhase(restorePD, metadata))
+
+	// Assert
+	restorePD, err = getRestorePrivateData(cfg.finRepo, cfg.restoreInputs[1].ActionUID)
+	require.NoError(t, err)
+	assert.Equal(t, Completed, restorePD.Phase)
+
+	// Verify the contents of the restore file.
+	restoreVolume, err := os.Open(cfg.restoreVol.GetPath())
+	require.NoError(t, err)
+	defer func() { require.NoError(t, restoreVolume.Close()) }()
+
+	expectedData := make([]byte, defaultVolumeSize)
+	copy(expectedData, cfg.volumeRaws[0][:defaultMaxPartSize])
+	copy(expectedData[defaultMaxPartSize:], cfg.volumeRaws[1][defaultMaxPartSize:])
+	utils.CompareReaders(t, bytes.NewReader(expectedData), restoreVolume)
+}
+
+func TestRestore_doRestoreDiffPhase_error(t *testing.T) {
+	// CSATEST-1589
+	// Description:
+	//   Confirm error processing for doRestoreDiffPhase.
+	//
+	// Arrange:
+	//   There is a full backup and incremental backup.
+	//   Restore destination volume is restored with the full backup.
+	//   The phase of private_data in the action_status table is set to `restore_diff` and
+	//   nextDiffPart is 0.
+	//
+	// Act:
+	//   Run doRestoreDiffPhase process about the incremental backup.
+	//
+	// Assert:
+	//   The process will return nil.
+	//   The phase of private_data in the action_status table is set to `completed`.
+	//   The contents of the restore volume have been restored.
+
+	// Arrange
+	cfg := setup(t, &setupInput{
+		enableIncrementalBackup: true,
+	})
+	r := NewRestore(cfg.restoreInputs[0])
+	require.NoError(t, r.Perform())
+
+	err := cfg.nlvRepo.RemoveDiffDirRecursively(cfg.restoreInputs[1].TargetSnapshotID)
+	require.NoError(t, err)
+
+	restorePD := &restorePrivateData{
+		Phase:        RestoreDiff,
+		NextDiffPart: 0,
+	}
+	createPrivateData(t, cfg, cfg.restoreInputs[1], restorePD)
+
+	// Act
+	r = NewRestore(cfg.restoreInputs[1])
+	metadata, err := job.GetBackupMetadata(cfg.finRepo)
+	require.NoError(t, err)
+	assert.Error(t, r.doRestoreDiffPhase(restorePD, metadata))
+
+	// Assert
+	restorePD, err = getRestorePrivateData(cfg.finRepo, cfg.restoreInputs[1].ActionUID)
+	require.NoError(t, err)
+	assert.Equal(t, RestoreDiff, restorePD.Phase)
+
+	// Verify the contents of the restore file.
+	restoreVolume, err := os.Open(cfg.restoreVol.GetPath())
+	require.NoError(t, err)
+	defer func() { require.NoError(t, restoreVolume.Close()) }()
+
+	utils.CompareReaders(t, bytes.NewReader(cfg.volumeRaws[0]), restoreVolume)
+}
+
+func TestRestore_doRestoreDiffPhase_skip(t *testing.T) {
+	// CSATEST-1590
+	// Description:
+	//   Confirm skip processing for doRestoreDiffPhase.
+	//
+	// Arrange:
+	//   There is a full backup and incremental backup.
+	//   Restore destination volume is restored with the full backup.
+	//   The phase of private_data in the action_status table is set to `completed`.
+	//
+	// Act:
+	//   Run doRestoreDiffPhase process about the incremental backup.
```

Comment:
CSATEST-1590 requires us to write dummy data to the destination volume. However, it seems that such process is not necessary to test the skip processing of doRestoreDiffPhase. Should we modify the CSATEST-1590 record?

Reply:
I previously updated CSATEST-1590, but it was incomplete. I have updated it again.